OneCommunityGlobal/HighestGoodNetworkApp
# Xiaow TimeEntryForm refactor

# Description
This is the refactor of TimeEntryForm, TimeLog and TimeEntry along with related files. 

## Related PRS (if any):
To test this frontend PR you need to checkout the #[668](https://github.com/OneCommunityGlobal/HGNRest/pull/668#issue-2054757181) backend PR.
…

## Main changes explained:
- Restructure format of files, favoring destructure assignment, redux state management, and code categorization arranged in a centralized manner
- Add WBS and task names in the project/task selection for both TimeEntryForm and TimeLog
- Add wbs in state.userProject reducer for better data handling
- Add wbsId and taskId to timeentry data for better data displaying
- Refactor event handler logic
- Disable outdated test files of these component, will overhaul those in the future
…

## How to test:
1. check into current branch
2. check into #[668](https://github.com/OneCommunityGlobal/HGNRest/pull/668#issue-2054757181) backend PR.
3. do `npm install` and `npm run start:local` to run this PR locally
4. Clear site data/cache
5. log as admin/vol user
6. Test the five use cases of TimeEntryForm below, try to break it.

## Screenshots or videos of changes:

## Note:
- userProfile, userProjects and userTask reducers are solely for storing data displaying on page (not for logged in user, this is important). 

- There are five use cases of TimeEntryForm component, and there are two boolean values passed in props that affect the behavior of TimeEntryForm: fromTimer (whether it is opened from Timer), edit(whether it is opened from editting an old timeentry).
   1. Account Owner (logged in user, not owner role account) user logs new time entry from Timer to Account Owner
   2. Account Owner adds time entry from 'Add Intangible Time Entry' button to Account Owner
   3. Account Owner edits same day time entry of Account user
   4. Account with permission (like owner or admin) adds time entry from 'Add Intangible Time Entry for XXX' button to XXX user
   5. Account with permission edits existing time entry of other user
 
- The recommended approach for these 5 cases are:
  1. Login as any user (better to use volunteer account, since it has the least permissions), start timer and log time, check and see if there is any issue in the process of logging time, and if you could see the newly added time entry on the Time Log page.
  2. Login as any user, use `Add Intangible Time Entry` button from Time Log page to add intangible time entry, check for issues and the newly added time entry on Time Log page. 
  3. Login as any user, edit the time entry that was logged in the same day, check for issues and the new changes on that time entry. For users with limited edit times (like volunteer user), check if you could see the warning modal for 5 edit times limits. Make sure other time entries that are logged in prior days don't have the edit button shown on that entry.
  4. Login as owner/admin account, open the Time Log page of other account (not logged in account, you would see a yellow banner say `You are currently viewing the dashboard/Timelog for XXX`), use the `Add Intangible Time Entry` button from other account's page to add intangible time entry for other account, check for issues and the newly added time entry on the other account's Time Log page (you need to log in that account and see from there).
  5. Login as owner/admin account, open the Time Log page of other account, edit the time entry from other account's page, check for issues and the new changes on that time entry.
